app/providers/atmosphere/optimization_strategy/default.rb

```diff
@@ -210,13 +220,13 @@ def get_optimal_flavor_for_tenant(tmpl, t)
         end
 
         def matching_flavors(tmpl, t)
-          fl = Atmosphere::VirtualMachineFlavor.arel_table
-          Atmosphere::VirtualMachineFlavor.active.where(
-            fl[:tenant_id].eq(t.id).and(fl[:memory].gteq(min_mem)).
-              and(fl[:cpu].gteq(min_cpu)).and(fl[:hdd].gteq(min_hdd)).
-              and(fl[:supported_architectures].
-                in([tmpl.architecture, 'i386_and_x86_64']))).
-          includes(flavor_os_families: :os_family)
+          t.virtual_machine_flavors.select do |vmf|
+              vmf.active &&
+              vmf.memory >= min_mem &&
+              vmf.cpu >= min_cpu &&
+              vmf.hdd >= min_hdd &&
+              ([tmpl.architecture, 'i386_and_x86_64'].include? vmf.supported_architectures)
```

Comment:
Line is too long. [91/80]<br>Use 2 (not 0) spaces for indenting an expression spanning multiple lines.